### ISSUE TITLE
Federation acceptance test fails with file/folder doesn't exist

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#33932]
I am seeing some CI fails like https://drone.owncloud.com/owncloud/core/13700/622
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/webUISharingExternal/federationSharing.feature:157
    /drone/src/tests/acceptance/features/webUISharingExternal/federationSharing.feature:169
    /drone/src/tests/acceptance/features/webUISharingExternal/federationSharing.feature:179

42 scenarios (39 passed, 3 failed)
413 steps (391 passed, 3 failed, 19 skipped)
23m2.30s (21.47Mb)
webUI test run failed with exit status: 1
```

```
  Scenario: receive same name federation share from two users                                           # /drone/src/tests/acceptance/features/webUISharingExternal/federationSharing.feature:157
    Given using server "REMOTE"                                                                         # FeatureContext::usingServer()
    And user "user2" has been created with default attributes                                           # FeatureContext::userHasBeenCreatedWithDefaultAttributes()
    And user "user1" from server "REMOTE" has shared "/lorem.txt" with user "user1" from server "LOCAL" # FederationContext::userFromServerHasSharedWithUserFromServer()
      Could not share file/folder! message: "Wrong path, file/folder doesn't exist"
      Failed asserting that two strings are equal.
      --- Expected
      +++ Actual
      @@ @@
      -'100'
      +'404'
```

I am raising this issue so that I can keep a record of when it happens, and then investigate what could have changed in the last few days.

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/9161 (Migration test still fails with "index ... already exists"), https://github.com/owncloud/core/issues/22485 (test), https://github.com/owncloud/core/issues/31579 (test), https://github.com/owncloud/core/issues/22492 (test), and https://github.com/owncloud/core/issues/33325 ([webUI tests] file delete acceptance tests can fail intermittently).

**Comment by phil-davis** [comment]
Similar fail here also: https://drone.owncloud.com/owncloud/files_primary_s3/720/340
```
--- Failed scenarios:

    /drone/server/tests/acceptance/features/webUISharingExternal/federationSharing.feature:157
    /drone/server/tests/acceptance/features/webUISharingExternal/federationSharing.feature:169
/drone/server/tests/acceptance/features/webUISharingExternal/federationSharing.feature:179
```

**Comment by phil-davis** [comment]
Should be fixed or improved by #34005
